xarray/core/dataarray.py

```diff
@@ -1588,9 +1590,10 @@ def func(self, other):
             other_variable = getattr(other, 'variable', other)
             other_coords = getattr(other, 'coords', None)
 
-            variable = (f(self.variable, other_variable)
-                        if not reflexive
-                        else f(other_variable, self.variable))
+            with np.errstate(all='ignore'):
```

Comment:
I don't think we need this here -- this should be caught at the level of the `Variable` wrapper.